### ISSUE TITLE
Is it possible to export the raw UML?

### ISSUE THREAD

**Issue by Midnighter** [jonatan-kazmierczak/class-visualizer#3]
First of all, this is a really nice tool to explore an unfamiliar project. Thank you for creating it!

I am aware that I can mark and then copy and paste the diagram from the preview panel but I was wondering if it is possible to export the UML code itself that was used to generate the preview? I would like to modify that further before generating my own visualization.

**Comment by jonatan-kazmierczak** [comment]
Hi, thanks!
There is no possibility to export UML model, the tool was not meant for that.
Could you please clarify, what exactly would you expect - what data (whole model / part of it) in what format (XMI will not be supported as it's structure doesn't match the internal model representation) ?  
Thank you.

BTW, I can recommend you the following article:
https://github.com/jonatan-kazmierczak/articles/blob/master/near-future.md#what-should-you-expect-in-the-near-future

**Comment by Midnighter** [comment]
I would love to be able to visualize with [PlantUML](https://plantuml.com/class-diagram) and thus to be able to get the model from the preview in the PlantUML specific syntax. I do realize that's a lot of work on your side, though, so I'm not expecting anything.

Whether to get the whole or part of the model and relationships with other classes depends a lot on the size of the class in terms of number of properties and methods. If it was possible to get output in the format for PlantUML, exporting everything first and then manually deleting parts is fine, I think.

> BTW, I can recommend you the following article:
https://github.com/jonatan-kazmierczak/articles/blob/master/near-future.md#what-should-you-expect-in-the-near-future

I expected a roadmap; not this. Quirky.

**Comment by jonatan-kazmierczak** [comment]
Thanks for the clarification!  
Export with members doesn't make sense IMO, but without - could be useful.  
I work with projects having >10k classes and a lot of relations. I cannot imagine anyone editing such export manually in a PlantUML file. I also cannot imagine a class diagram with all such classes :)  

What could make sense, is export of a subset of classes - i.e. the ones selected on the classes list.
Just that it's sometimes not easy to select all relevant classes - especially when there are thousands of them on the list.  
How do you see that?

**Comment by Midnighter** [comment]
I agree, with so many classes it doesn't make sense to export all of them.

It could be nice to be able to select a number of them from the 'Hierarchy' or even make a visual selection in the 'Relations Diagram'.

**Comment by jonatan-kazmierczak** [comment]
> It could be nice to be able to select a number of them from the 'Hierarchy' or even make a visual selection in the 'Relations Diagram'.

- on *Hierarchy* there are no interfaces and it cannot be filtered - not very useful for selection
- on *Relations Diagram* only 1 element can be selected at a time - technical decision

*List* seems to be the easiest starting point. User can even select the whole list and export everything.  
It is also possible to export all members and hide them by default in PlantUML.

I see the feature as follows from the user perspective:
- select classes on the *List*
- choose new option *Export as PlantUML", following by the "Save as" dialog
- choose the name of the PlantUML file

How do you see that?

**Comment by Midnighter** [comment]
Your assessment sounds good to me and I'm really happy that you're considering this feature!

**Comment by jonatan-kazmierczak** [comment]
**Summary**

The proposed feature looks as follows from the user perspective:
- select classes on the *List*
- choose new option *Export as PlantUML", following by the "Save as" dialog
- choose the name of the PlantUML file
- selected classes with relations are saved in the file  
  (optionally: with members, which should be hidden by default)

I can take care of the feature possibly at the end of 2020, but I can support committers, which would like to contribute to the feature earlier.
